### PR TITLE
feat: cloudwatch steps

### PR DESCRIPTION
**Pull request by Nelsonochoam**
**Description of changes:**
This was created off of https://github.com/aws-greengrass/aws-greengrass-log-manager/compare/zeel-nelson-cloudwatch-steps?expand=1. This changes contain new cloud-watch steps to verify if a stream got created and also has the logic for removing the created resources once the test finish running.

This PR also requires a new step in OTF https://github.com/aws-greengrass/aws-greengrass-testing/pull/154

### CODE REVIEW

File: uat/src/main/resources/greengrass/features/log-manager-1.feature [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
 Feature: Greengrass V2 LogManager
 
-  As a customer, I want to selectively upload my logs and metrics to AWS Cloudwatch to save cost.
+    As a customer, I want to selectively upload my logs and metrics to AWS Cloudwatch to save cost.
 
-  Background:
-    Given my device is registered as a Thing
-    And my device is running Greengrass
-    And 5 temporary rotated log files for component aws.greengrass.Nucleus
-    And 5 temporary rotated log files for component aws.greengrass.StreamManager
```
Comment (dgggIoT) [inline uat/src/main/resources/greengrass/features/log-manager-1.feature:10]: Shall we still include the plugin component? Although we don`t really use it, but people may question if the LM is compatible with other plugin component.

Comment (Nelsonochoam) [reply uat/src/main/resources/greengrass/features/log-manager-1.feature:10]: Basically trying to copy pasta, not trying to change the tests yet

**Review by dgggIoT — DISMISSED** [review]
Just some minor comments. But overall changes are good, so it is approved.

File: uat/src/main/resources/greengrass/features/log-manager-1.feature [hunk 2]
```diff
@@ -31,6 +30,7 @@ Feature: Greengrass V2 LogManager
...
         }
         """
-    And I deploy the Greengrass deployment configuration
-    Then the Greengrass deployment is COMPLETED on the device after 5 minutes
-    And I verify the aws.greengrass.LogManager component is RUNNING using the greengrass-cli
\ No newline at end of file
+        And I deploy the Greengrass deployment configuration
+        Then the Greengrass deployment is COMPLETED on the device after 2 minutes
+        And I verify the aws.greengrass.LogManager component is RUNNING using the greengrass-cli
+        And I verify that it created a log group for component type GreengrassSystemComponent for component System, with streams within 120 seconds in CloudWatch
```
[the remark this answers is not in the document]
Comment (Nelsonochoam) [reply uat/src/main/resources/greengrass/features/log-manager-1.feature:36]: Yeah... again, not trying to expand the test just preserve its functionality.... there will be other PRs adding that in.

File: uat/pom.xml [hunk 3]
```diff
@@ -23,22 +21,64 @@
...
             <groupId>com.aws.greengrass</groupId>
             <artifactId>aws-greengrass-testing-standalone</artifactId>
             <version>1.0-SNAPSHOT</version>
             <scope>compile</scope>
         </dependency>
-
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>cloudwatch</artifactId>
+            <version>2.18.13</version>
```
Comment (urvashijain18) [inline uat/pom.xml:34]: OTF is using [sdk version](https://github.com/aws-greengrass/aws-greengrass-testing/blob/main/pom.xml#L40): 2.17.254
Just pointing out version differences.

**Review by urvashijain18 — CHANGES_REQUESTED** [review]

File: uat/src/main/java/com.aws.greengrass/resources/CloudWatchLogStreamSpecModel.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        // This method doesn't actually create a CloudWatch stream. It just registers the resource as
+        // created do it can get deleted during teardown.
+        CloudWatchLogStream stream = CloudWatchLogStream.builder()
+                .groupName(logGroupName())
+                .streamName(logStreamName())
+                .build();
+
+        return CloudWatchLogStreamSpec.builder()
+                .from(this)
+                .created(true)
```
[the remark this answers is not in the document]
Comment (Nelsonochoam) [reply uat/src/main/java/com.aws.greengrass/resources/CloudWatchLogStreamSpecModel.java:30]: Yeah... if we don't tag it as created it doesn't get deleted

**Review by dgggIoT — APPROVED** [review]

**Pull request merged**